test/Broadway/EventHandling/SimpleEventBusTest.php

```diff
@@ -104,6 +104,41 @@ public function it_does_not_dispatch_new_events_before_all_listeners_have_run()
         $this->eventBus->publish($domainEventStream);
     }
 
+    /**
+     * @test
+     */
+    public function it_should_still_publish_events_after_exception()
+    {
+        $domainMessage1 = $this->createDomainMessage(array('foo' => 'bar'));
+        $domainMessage2 = $this->createDomainMessage(array('foo' => 'bas'));
+
+        $domainEventStream1 = new DomainEventStream(array($domainMessage1));
+        $domainEventStream2 = new DomainEventStream(array($domainMessage2));
+
+        $eventListener = $this->createEventListenerMock();
+        $eventListener
+            ->expects($this->at(0))
```

Comment:
In this case it would be acceptable, but normally its better not make your test depend on the internal flow of a method your are testing.

The "problem" with PHPUnit mocks is that every method call made on a mock with `at()` will make the mock expect exactly that method call at that time. But if you change the calling order (later/sooner) then the test will fail the expectation. So every time you do a refactor the tests need to be updated and can't be safely used to test if you broke something.

As this interface has only one method its less fragile, and therefor less likely to break.
But if you have an interface with multiple methods, and you add expectations for all of them in a single test its going to break someday.